src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -14,12 +14,188 @@
 #include "error_codes.h"
 #include "deps_format.h"
 
-pal::string_t fx_muxer_t::resolve_fx_dir(const pal::string_t& muxer_dir, runtime_config_t* runtime)
+pal::string_t resolve_impl_version_from_deps(const pal::string_t& deps_json)
 {
-    trace::verbose(_X("--- Resolving FX directory from muxer dir [%s]"), muxer_dir.c_str());
-    const auto fx_name = runtime->get_fx_name();
-    const auto fx_ver = runtime->get_fx_version();
-    const auto patch_roll_fwd = runtime->get_patch_roll_fwd();
+    trace::verbose(_X("--- Resolving %s version from deps json [%s]"), LIBHOSTPOLICY_NAME, deps_json.c_str());
+
+    pal::string_t retval;
+    if (!pal::file_exists(deps_json))
+    {
+        trace::verbose(_X("[%s] does not exist"), deps_json.c_str());
+        return retval;
+    }
+
+    pal::ifstream_t file(deps_json);
+    if (!file.good())
+    {
+        trace::verbose(_X("[%s] could not be opened"), deps_json.c_str());
+        return retval;
+    }
+
+    if (skip_utf8_bom(&file))
+    {
+        trace::verbose(_X("UTF-8 BOM skipped while reading [%s]"), deps_json.c_str());
+    }
+
+    try
+    {
+        const auto root = json_value::parse(file);
+        const auto& json = root.as_object();
+        const auto& libraries = json.at(_X("libraries")).as_object();
+
+        pal::string_t prefix = _STRINGIFY(HOST_POLICY_PKG_NAME) + pal::string_t(_X("/"));
+        for (const auto& library : libraries)
+        {
+            if (starts_with(library.first, prefix, false))
+            {
+                retval = library.first.substr(prefix.size());
+                break;
+            }
+        }
+    }
+    catch (const std::exception& je)
+    {
+        pal::string_t jes;
+        (void)pal::utf8_palstring(je.what(), &jes);
+        trace::error(_X("A JSON parsing exception occurred in [%s]: %s"), deps_json.c_str(), jes.c_str());
+    }
+    trace::verbose(_X("%s version is [%s] in deps json file [%s]"), retval.c_str(), deps_json.c_str());
+    return retval;
+}
+
+bool to_hostpolicy_package_path(const pal::string_t& dir, const pal::string_t& version, pal::string_t* candidate)
+{
+    candidate->clear();
```

Comment:
De-referencing a pointer passed into a function without asserting it is non-null; please add an assert != nullptr